commands/shooter_test.py

```diff
@@ -8,12 +10,31 @@ class ShooterTest(Command):
     def __init__(self):
         super().__init__(subsystem=Command.getRobot().shooter, name="Shooter")
 
+        self.integral = 0
+        self.previous_error = 0
+
+        self.start_speed = 0
+        self.goal_speed = Constants.SHOOTER_SPEED_PERCENT_OUTPUT
+        self.top_motor_current_speed = Command.getRobot().shooter.top_shooter_velocity
+        self.bottom_motor_current_speed = Command.getRobot().shooter.bottom_shooter_velocity
+
+        top_error = self.goal_speed - self.top_motor_current_speed
+        bottom_error = self.goal_speed - self.bottom_motor_current_speed
+        self.top_integral = self.integral + top_error
+        self.bottom_integral = self.integral + bottom_error
+        self.top_derivative = (top_error - self.previous_error)
+        self.bottom_derivative = (bottom_error - self.previous_error)
+        self.top_output = Constants.SHOOTER_VELOCITY_P * top_error + Constants.SHOOTER_VELOCITY_I * self.top_integral + Constants.SHOOTER_VELOCITY_D * self.top_derivative
+        self.bottom_output = Constants.SHOOTER_VELOCITY_P * bottom_error + Constants.SHOOTER_VELOCITY_I * self.bottom_integral + Constants.SHOOTER_VELOCITY_D * self.bottom_derivative
+
```

Comment:
Love that you did this, but this will run on the Roborio (which is the computer on the robot). We want this to run on the talon's, which is referred to as 'hardware PID'. Basically you don't need any of this and can accomplish it in about 2 lines of code. Good initiative though. Also, the plan right now is not to use CIM motors (which use talonSRX's) and instead use NEO motors (which use the Spark Max motor controller). These use a slightly different method for control, so check out [the docs](http://www.revrobotics.com/content/sw/max/sw-docs/java/com/revrobotics/CANPIDController.html) and [some examples](https://github.com/REVrobotics/SPARK-MAX-Examples/tree/master/Java/Velocity%20Closed%20Loop%20Control/src/main/java/frc/robot). These are written in Java, but try to read through them and check with me if you need help. Also, most of the configuration should be done in the subsystem, not the command.